OneCommunityGlobal/HighestGoodNetworkApp
# yihan create submit for review button

# Description
<img width="719" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/4536b0b8-dc49-4dc8-8e9d-04299c0ecd26">

## Related PRS (if any):
This frontend PR is related to the [#443 ](https://github.com/OneCommunityGlobal/HGNRest/pull/443)backend PR.

## Main changes explained:
- Update file TeamMemberTask.jsx for including ReviewButton component, and transfering userId and updateTask function to it.
- Update file TeamMemberTasks.jsx for transfering userId to TeamMemberTask
- Create file ReviewButton.jsx for introducing review button.
- Create file reviewButton.css for formatting review button

## How to test:
1. check into current branch on both frontend and backend.
2. do `npm install` and `...` to run this PR locally
3. log as any user (make sure the user has task assigned and all the accounts you use for this test are in same team)
4. go to dashboard → Tasks, verify only the user who is assigned the task can see "submit for review".
5. Click on "submit for review" -> Submit, see if the button turns to
a. "Ready for review" dropdown for Owners/Admins/Managers/Mentors who can see this user in their tasks list.
b. "More work needed, reset this button" to the user who is assigned the task (only for volunteer and core team user)
c. disabled "Ready for review" button for other type account who can see this user in their tasks list.
6. Click on "More work needed, reset this button" button, see it reset to "Submit for Review".
7.  repeat step 5.
8. go to "Ready for review" dropdown -> “More work needed, reset this button”, the dropdown and disabled button should disappear, and "submit for review" button shows again (same as step 6).
8. repeat step 5
9. go to "Ready for review" dropdown -> “✓ as complete and remove task”, see if the task is removed from task list.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/f1df014a-5a0e-4f56-9e2d-49633b1853b9



## Note:
It may take some time for button to update.
